Add Details to Skeletal Code

joytqt-1202 opened the pull request:
# Duke

### Bullet List
* Add Todo, Event, Deadline Functionality into ChatBot
    * Add Marking Functionality
* Add Delete Functionality
---------------
### Numbered List
1. first
    1.1. second-first?
3. second
4. third
---------------
### Fenced Code Block
1. `without` syntax highlighting
```
public class fencedCodeBlock {
    private String please;
    /** Creates a fenced code block -- not really*/
    public fencedCodeBlock() {
        this.please = "NO";
    }
}
```
2. `with` syntax highlighting :)
```java
public class fencedCodeBlock {
    private String please;
    /** Creates a fenced code block -- not really*/
    public fencedCodeBlock() {
        this.please = "NO";
    }
}
```
---------------
### Task list
- [X] DUKE!!
- [ ] why are you using this instead of [my duke](www.why_not_duke.com)?
    1. you will be happy
    2. duke will be happy
        - i will be happy
- [ ] explanation please
---------------
### Emojis?!
😀  😄  😢  💯  😭 <br>
i like this [emoji cheatsheet](https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md)
i find it *pretty* **useful**
---------------
### Block quote 
> "you *mean* ~Duke~ **me**?"

Commit:
Add Gradle support

joytqt-1202 committed:
Level-1. Skeletal version of Duke (Name of chatbot to be changed)

joytqt-1202 committed:
Level-2. Adding items into list and displaying list.

joytqt-1202 committed:
Level-2. Reverted back to if-else statements

joytqt-1202 committed:
Level 3. adding mark and unmark commands

joytqt-1202 committed:
Level-4. Categorise tasks into Deadlines, Events and Todos.

Changed implementation of command calling from if-else to switch case

joytqt-1202 committed:
Updated inheritance for Tasks

joytqt-1202 committed:
Tried Automated TextUiTesting.

joytqt-1202 committed:
Level-5. Handling Errors (Basic)
Only included minimal error handling.

Warnings yet to be included:
- incorrect number of parameters (/) for task adding
- marking a task number that is longer than the list length
- marking an already marked task (and for unmarking)

joytqt-1202 committed:
Level-6. New Delete Function

Add support for deleting functions from the list.

New warning to implement:
- deleting a task that does not exist.

joytqt-1202 committed:
A-Enums. Use Java enum for jobs

hansstanley commented:
@joytqt-1202 Perhaps you could rename the PR to `[joytqt-1202] iP` to match the instructions?

> Set the PR name as [{Your full/partial name or your Github username}] iP
e.g., [Richard Mathews Chee] iP or [Rich ... hee] iP or [TheRichMat] iP

Also, the compiled `.class` files are usually not committed to the repo. If you use IntelliJ, building the program should put all the `.class` files in a `./out/` folder, which has already been ignored by git with its inclusion in the `.gitignore` file.

hansstanley commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,81 @@
+            job = Job.INVALID;
+        }
+
+        switch(job) {
+            case LIST:
+                Task.listTasks();
+                break;
+            case MARK:
+                Task.markTasks(Integer.parseInt(commands[1])-1);
+                break;
+            case UNMARK:
+                Task.unmarkTasks(Integer.parseInt(commands[1])-1);
+                break;
+            case DELETE:
+                Task.deleteTask(Integer.parseInt(commands[1])-1);
+                break;
+            default:
+                String taskName = "";
+                for (int i = 1; i < commands.length; i++) {
+                    taskName += commands[i] + " ";
+                }
+                try {
+                    Task.addTask(commands[0], taskName);
+                } catch(Exception e) {
+                    System.out.println(e.getMessage());
+                }
+        }
```
Could you remove the indentation for `case` statements to match the coding standard?

For e.g.,

```java
switch (job) {
case LIST:
    // ...
case MARK:
    // ...
// ...
}
```

hansstanley commented on src/main/java/Tasks/Event.java:
```diff
@@ -0,0 +1,21 @@
+
+    @Override
+    public String toString() {
+        return "[E] " + this.TasktoString()
+            + "( from: " + this.startTime 
+            + ") ( to: " + this.endTime + ")";
```
Perhaps increase the indentation of wrapped lines to 8 spaces to match the coding standard?

```suggestion
        return "[E] " + this.TasktoString()
                + "( from: " + this.startTime 
                + ") ( to: " + this.endTime + ")";
```

hansstanley commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,73 @@
+
+import java.util.ArrayList;
+import exceptions.*;
+
```
This class could use a JavaDoc to explain its purpose.

hansstanley commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,73 @@
+        return this.isChecked ? "[X]" : "[ ]";
+    }
+
+    protected String TasktoString() {
```
Method names are usually in lowerCamelCase, i.e. `taskToString`.

Alternatively, you could override the `toString` method.

hansstanley reviewed (commented):
Just a few coding standard comments, keep it up!

joytqt-1202 committed:
Level-7: Save generated list into txt file

List generated will now save into a text file which can be viewed even after chat has ended

Other Relevant Info:
Incorporate more enums for tasks
Update code to follow standard Java Coding Conventions

joytqt-1202 committed:
Level 8: Dates and Times

Understanding dates and times

joytqt-1202 committed:
Level 8: Update Dates and Times for Deadlines

Previously, no restriction was placed on deadline and event date and timings.

Now, user would have to enter date and timings in the format: yyyy-mm-ddThh:mm
and the user can check for deadlines due on a certain date or events that are ongoing on that date

joytqt-1202 committed:
Merge branch 'branch-level-8'

Resolve conflicts between branch-Level-8 and branch-Level-7

joytqt-1202 committed:
Merge remote-tracking branch 'origin' into branch-Level-8

joytqt-1202 committed:
Merge branch 'branch-Level-8'

joytqt-1202 committed:
A-MoreOOP

Make the Code more OOP
Refactor the code to extract out closely related code as classes

joytqt-1202 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

joytqt-1202 committed:
Storage Loading: enable loading of previous list

Due to changes made in A-MoreOOP, issues were generated for loading previous list.

Adjustment made to parse date value in stored previous list

Previous list dates are stored in "MMM dd yyyy, DAY_OF_WEEK, hh:mmAM/PM" rather than in "yyyy-MMM-ddTHH:mm", which is the required format of users when entering data.

This parsing method could be used to allow users to enter dates in the format that is stored in the lists, rather than in the current required format.  Might make this available in future commits.

joytqt-1202 committed:
Parsing: fix parsing error for check command

Error thrown when user calls check command

Update parsing error

joytqt-1202 committed:
Gradle: A-Gradle

Automate project builds using Gradle

Build and run Duke manually by using javac and java
Check style of code also done manually

Merge Gradle branch to automate build, run and checkstyle

Automation allows for simpler and faster builds

joytqt-1202 committed:
Add JUnit Testing, Update Mark Command

The following things have been added in this commit
* Add JUnit tests to test the behaviour of certain parts of code.
* Update Mark Command index to reflect expected behaviour.

Testing was manually by running Duke and performing tests.
Mark Command followed 0-based indexing, rather than the  expected 1-based indexing.

Incorporating JUnit Testing to allow fast and easy testing of previous codes
to ensure that code is not broken after new updates.
People would tend to count tasks using 1-indexing, rather than 0-indexing.

joytqt-1202 committed:
Update Exception Catching for Task Creation

Update ArrayOutOfBoundsException in Task.java to NullPointerException
to catch the appropriate error thrown for insufficient arguments supplied

Insufficient arguments error not caught, leads to exception thrown
and early termination of program

joytqt-1202 committed:
Update condition and add Java docs in exceptions

Update conditional statements in InsufficientArgumentException.java.
Add java docs in exception files for documentation.

Insufficient Argument Exception could be used incorrectly.
Missing java docs

Better error catching and Java docs for documentation

Earlier commit also had additions of java docs.

joytqt-1202 committed:
Add javadocs for parser package files

joytqt-1202 committed:
Add javadocs for parser package files

joytqt-1202 committed:
Add javadocs for storage package files

joytqt-1202 committed:
Add Javadocs for tasks package

joytqt-1202 committed:
Add Javadocs for ui package

joytqt-1202 committed:
Tweak code to comply with java coding standard

joytqt-1202 committed:
Add Find Functionality

Give users a new way to find a task by searching for a keyword.

Users only able to filter tasks using check command.

Users may want to filter their tasks based on keywords to look
through a smaller list of relevant tasks

Find Command created to give users a find a task functionality
so that they can search using a keyword to view filtered tasks
that contain the specified keyword

Merged CheckCommand into FindCommand since they have
simlar functionalities

joytqt-1202 committed:
Level-10: GUI

Implement JavaFX basics. Not yet applied to Duke

To use Duke, users are provided with the command line interface
which is not really user friendly and looks boring

Use JavaFX to introduce graphical user interface for users to
interact with, rather than a plain command line interface

Upcoming: GUI for Duke

joytqt-1202 committed:
Level-10: GUI Updates

Duke original functionaity can be performed on graphical user interface

Bare GUI implementation. No support for Duke's functionality.

We want Duke's functionality to be made available.

joytqt-1202 committed:
Assertions

Add assertion in Parser file

Unnecessary throw code.

May cause confusion in debugging.

joytqt-1202 committed:
Update code to improve code quality

joytqt-1202 committed:
Merge pull request #2 from joytqt-1202/branch-A-Assertions

Assertions

joytqt-1202 committed:
Merge pull request #3 from joytqt-1202/branch-A-CodeQuality

Update code to improve code quality

joytqt-1202 committed:
Merge branch 'branch-A-CodeQuality'

joytqt-1202 committed:
Remove redundant code

joytqt-1202 committed:
Add an extension: B-Reminders

Add reminder functionality to remind users of either deadline tasks with a
deadline of the current day or event tasks that span over the current day.

Users have to search through their list on their own to view the such tasks.
They may forget to do so and miss out on tasks with deadlines

* Use Find Functionality upon initialisation of application to remind users
of such tasks
* Update Invalid Format Exception - Use of Java Varargs for Exceptions thrown
for Find Command
* Improve Code Quality

Reminders - Reminds users of such tasks upon starting the application so that they
can perform the various tasks required of them, before a task is overdue.
Update Exception - Exception is thrown for Find, Check Command, as well as
Add Command. Add Command requires LocalDateTime, while Find, Check Command can
accept both LocalDateTime and LocalDate.

Changes were also made to DialogBox.fxml to fix dialog box overflow

joytqt-1202 committed:
Merge branch 'master' of https://github.com/joytqt-1202/ip

joytqt-1202 committed:
Merge branch 'branch-A-JavaDoc' of https://github.com/joytqt-1202/ip

joytqt-1202 committed:
Fix Invalid Date Format Exception

joytqt-1202 committed:
Update UI dimensions to acocomodate for overflow of text

joytqt-1202 committed:
Fix grammatical errors, Update UI

joytqt-1202 committed:
Update Code to comply with Java CheckStyle

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Fix Grammatical Errors

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Fix bolding in README.md

joytqt-1202 committed:
A-BetterGui: Improve the GUI

Improve the Graphical User Interface to make it more polished

User is limited to using Duke with a certain application window size.

Allow resizing of the Window and have the content resize accordingly

joytqt-1202 committed:
Improve termination of application

Stop user from providing further inputs into Duke

Users could still continue to provide inputs, just that
the task list is not saved.

Remove input text field and send button so user is unable
to further communicate with Duke

Further development:
* automatically close application after 10s
    -- let user know that the application will close after 10s
* let the user know to relaunch Duke to continue
communications with Duke

joytqt-1202 committed:
Update to autoclose after saying "bye"

After user says "bye", Duke says bye.
Application window stays open until user manually closes it.

Update to change duke's response to bid user goodbye and let
user know that the application window will close in 10 seconds
Make the application window close automatically
10 seconds after user says "bye" to Duke

Prevent causing confusion in user due to application window
being left open after saying bye.
User would tend not to need any previous details after saying
bye to duke and would expect the application to close

joytqt-1202 committed:
Update storage file path

joytqt-1202 committed:
Remove auto close due to crashes during testing, Update display pictures

joytqt-1202 committed:
Update bye message

joytqt-1202 committed:
Fix Autoclose Application after bye

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md

joytqt-1202 committed:
Update README.md
